frontend/catalyst/jax_tracer.py

```diff
@@ -849,23 +850,19 @@ def trace_quantum_measurements(
         if isinstance(o, MeasurementProcess):
 
             # Check if the measurement is supported shot-vector where num_of_total_copies > 1
-            if device.shots.num_copies > 1 and o.return_type.value != "sample":  # qml.sample()
+            if device.shots.num_copies > 1 and not isinstance(o, qml.measurements.SampleMP):
                 raise NotImplementedError(
-                    f"Measurement {o.return_type.value} is not supported a shot-vector. "
+                    f"Measurement {type(o).__name__} is not supported a shot-vector. "
                     "Use qml.sample() instead."
                 )
 
-            if isinstance(device, qml.devices.LegacyDevice):
-                m_wires = o.wires if o.wires else range(device.num_wires)
-            else:
-                m_wires = o.wires if o.wires else range(len(device.wires))
+            m_wires = o.wires if o.wires else range(len(device.wires))
 
             obs_tracers, nqubits = trace_observables(o.obs, qrp, m_wires)
 
             using_compbasis = obs_tracers.primitive == compbasis_p
 
-            if o.return_type.value == "sample":
-                results = []  # list of results per copy
+            if isinstance(o, qml.measurements.SampleMP):
```

Comment:
Since we discussed this for StateMP, would it not be better to check the type exactly for all of them (`type(o) is SampleMP` etc)?

Reply:
Done.